common/lib/utils.cpp

```diff
@@ -351,5 +355,35 @@ bool isPrivileged() {
   return true;
 }
 
+void write_to_file(std::string data, const std::filesystem::path file) {
+  villas::Log::get("Filewriter")->debug("{} > {}", data, file.u8string());
+  std::ofstream outputFile(file.u8string());
+
+  if (outputFile.is_open()) {
+    outputFile << data;
+    outputFile.close();
+  } else {
+    throw std::filesystem::filesystem_error("Cannot open outputfile",
+                                            std::error_code());
+  }
+}
+
+std::vector<std::string> read_names_in_directory(const std::string &name) {
+  DIR *directory = opendir(name.c_str());
+
+  struct dirent *dp;
+  std::vector<std::string> names;
+  dp = readdir(directory);
```

Comment:
Please be consistent in the APIs you are using.

I think instead of libc, you should use [`std::filesystem::directory_iterator`](https://en.cppreference.com/w/cpp/filesystem/directory_iterator) here..